src/all/komga/src/eu/kanade/tachiyomi/extension/all/komga/KomgaFilters.kt

```diff
@@ -0,0 +1,378 @@
+
+/**
+ * Filters for the [Komga] extension. The class' responsibility is observe the
+ * preferences for changes and fetch or clear the collections for the filters based
+ * on the new state. The observable is throttled by one second.
+ */
+class KomgaFilters(
+    private val client: OkHttpClient,
+    private val headers: Headers,
+    private val preferences: SharedPreferences,
+    private val preferencesSubject: BehaviorSubject<String?>,
+    private val onError: (message: String, e: Throwable?) -> Unit
+) {
+
+    private val baseUrl: String
+        get() = preferences.getString(Komga.ADDRESS_TITLE, Komga.ADDRESS_DEFAULT)!!
+
+    private val username: String
+        get() = preferences.getString(Komga.USERNAME_TITLE, Komga.USERNAME_DEFAULT)!!
+
+    private val password: String
+        get() = preferences.getString(Komga.PASSWORD_TITLE, Komga.PASSWORD_DEFAULT)!!
+
+    private var libraries = emptyList<LibraryDto>()
+    private var collections = emptyList<CollectionDto>()
+    private var genres = emptySet<String>()
+    private var tags = emptySet<String>()
+    private var publishers = emptySet<String>()
+
+    // Roles to list of authors
+    private var authors = emptyMap<String, List<AuthorDto>>()
+
+    private val json: Json by injectLazy()
+
+    init {
+        observePreferencesForFilters()
+    }
+
+    private fun observePreferencesForFilters() {
+        preferencesSubject
+            .subscribeOn(Schedulers.io())
+            .observeOn(Schedulers.io())
+            .throttleWithTimeout(1, TimeUnit.SECONDS)
+            .subscribe({ fetchCollectionsForFilters() }) {
+                onError("error while fetching collections for filters", it)
+            }
+    }
+
+    private fun fetchCollectionsForFilters() {
+        if (baseUrl.isBlank() || username.isBlank() || password.isBlank()) {
+            libraries = emptyList()
+            collections = emptyList()
+            genres = emptySet()
+            tags = emptySet()
+            publishers = emptySet()
+            authors = emptyMap()
+
+            return
+        }
+
+        fetchLibraries()
+        fetchCollections()
+        fetchGenres()
+        fetchTags()
+        fetchPublishers()
+        fetchAuthors()
+    }
+
+    private fun fetchLibraries() {
+        val librariesResult = runCatching {
+            val request = GET("$baseUrl/api/v1/libraries", headers)
+            client.newCall(request).execute().parseAs<List<LibraryDto>>()
+        }
+
+        if (librariesResult.isFailure) {
+            onError("error while loading libraries for filters", librariesResult.exceptionOrNull())
+        }
+
+        libraries = librariesResult.getOrDefault(emptyList())
+    }
+
+    private fun fetchCollections() {
+        val collectionsResult = runCatching {
+            val request = GET("$baseUrl/api/v1/collections?unpaged=true", headers)
+            client.newCall(request).execute().parseAs<List<CollectionDto>>()
```

Comment:
```suggestion
            client.newCall(request).execute().parseAs<PaginatedResponseDto<CollectionDto>>().content
```

this fixes the collection filter
type-safety kinda didn't help here. It got type cast? 🤔 
anyways, should be PaginatedResponse (was previously [PageWrapperDto](https://github.com/tachiyomiorg/tachiyomi-extensions/blob/master/src/all/komga/src/eu/kanade/tachiyomi/extension/all/komga/Komga.kt#L542))

could make it slightly more readable by having a separate variable (eg: outputList.content)
also don't forget: import PaginatedResponse from dto package

Reply:
Thanks for the heads-up, will fix soon 👍🏻